Game/Source/GDKTestGyms/Private/UserExperienceComponent.cpp

```diff
@@ -76,6 +82,23 @@ void UUserExperienceComponent::OnClientOwnershipGained()
 	GetWorld()->GetTimerManager().SetTimer(Timer, Delegate, 1.0f, true);
 }
 
+float UUserExperienceComponent::CalculateAverageVL() const
+{
+	const float NCDSquared = GetOwner()->NetCullDistanceSquared;
+
+	float Avg = 0.0f;
+	for (int i = 0; i < UpdateRate.Num(); i++)
+	{
+		const UpdateInfo& UR = UpdateRate[i];
+		check(UR.DistanceSq < NCDSquared);
+		// Reduce the impact of values that were received by up to half 
+		float Weight = 1.f - (UR.DistanceSq / NCDSquared)*0.5f;
```

Comment:
I suggest we sum all `(UR.DistanceSq / NCDSquared)` and then divide that here instead, this way we could still reason about the average in seconds (albeit weighted)